selfservice/strategy/code/code_recovery.go

```diff
@@ -0,0 +1,125 @@
+package code
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	"github.com/gofrs/uuid"
+
+	"github.com/ory/herodot"
+	"github.com/ory/x/randx"
+
+	"github.com/ory/kratos/identity"
+)
+
+type RecoveryCodeType int
+
+const (
+	RecoveryCodeTypeAdmin RecoveryCodeType = iota + 1
+	RecoveryCodeTypeSelfService
+)
+
+var (
+	ErrCodeNotFound    = herodot.ErrNotFound.WithReasonf("unknown recovery code")
+	ErrCodeAlreadyUsed = herodot.ErrNotFound.WithReasonf("recovery code was already used")
```

Comment:
Should this really be `404`? Not rather `Bad Request` or `Precondition Failed`?

Reply:
Right, I like bad request. Done.